### PR TITLE
alwayslink=1 for reflection plugin

### PR DESCRIPTION
**Pull request by iancoolidge**

### CODE REVIEW

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by nicolasnoble** [comment]
This is okay to test.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by nicolasnoble** [comment]
This is ok to test.

**Comment by iancoolidge** [comment]
I rebased this PR.

**Comment by iancoolidge** [comment]
Split this to two CLs. One to propagate alwayslink, and one to enable it for the reflection plugin (improve granularity)

**Comment by ncteisen** [comment]
Linking in @y-zeng since he is handling import. I _think_ this change is blocking some internal things (is that correct @iancoolidge?), so getting it in sooner would be nice

**Comment by iancoolidge** [comment]
We were told to commit this here first,
This fixes grpc_cli

**Comment by y-zeng** [comment]
Jenkins: test this please

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

File: bazel/grpc_build_system.bzl [hunk 1]
```diff
@@ -41,6 +42,7 @@ def grpc_cc_library(name, srcs = [], public_hdrs = [], hdrs = [],
     includes = [
         "include"
     ]
```
Comment (y-zeng) [inline bazel/grpc_build_system.bzl]: Missing a `,`

Comment (iancoolidge) [inline bazel/grpc_build_system.bzl]: D'oh! Thanks. Done.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by iancoolidge** [comment]
I pushed a new version of the branch that fixes that missing comma,
is there anything else I need to do??>

**Comment by y-zeng** [comment]
Jenkins: test this please

**Comment by iancoolidge** [comment]
Is this ready to merge?

**Comment by dgquintas** [comment]
Issues: #11109

**Pull request merged**